clang/test/CXX/drs/dr28xx.cpp

```diff
@@ -58,3 +58,24 @@ void B<int>::g() requires true;
 #endif
 
 } // namespace dr2847
+
+namespace dr2858 { // dr2858: 19
+
+#if __cplusplus > 202302L
+
+template<typename... Ts>
+struct A {
+  // FIXME: The nested-name-specifier in the following friend declarations are declarative,
```

Comment:
Doesn't this mean that we only partially implement the DR?

Reply:
Not calling `diagnoseQualifiedDeclaration` for friend declarations is unrelated to this DR. Besides that, the only reason we can't write a test to produce this diagnostic is because we compute the `DeclContext` of the _nested-name-specifier_ _before_ `diagnoseQualifiedDeclaration` is called.